### ISSUE TITLE
Setting Syncthing environment variable `all_proxy`

### ISSUE THREAD

**Issue by jiajunhuang** [xor-gate/syncthing-macosx#98]
how can I use proxy in `syncthing-macos`? I can set a `all_proxy=...` in systemd service file, but it does not provide a way to set environment variable in syncthing-macos.

**Comment by xor-gate** [comment]
Currently setting a proxy is not possible. So it is a missing feature.

**Comment by opensource21** [comment]
As a workaround you could follow https://github.com/syncthing/syncthing-macos/issues/154#issuecomment-832946496

**Comment by xor-gate** [comment]
@opensource21 setting environment variables is not possible with #154 only commandline parameters

**Comment by opensource21** [comment]
@xor-gate This why I link to the workaround where a shell-script is called. Otherwise, I had written it's solved with #154

**Comment by AlliotTech** [comment]
It seems we still need to use a shell script to override the entry point to use proxy.

**Comment by AlliotTech** [comment]
Here is a less elegant way to achieve the goal:
export all_proxy=socks5://127.0.0.1:1234 && open /Applications/Syncthing.app;

**Comment by xor-gate** [comment]
Yeah I understand you need to bypass the (corporate) firewall or proxy via e.g Tor using the SOCKS5 protocol. For my own reference we need to inject the [`all_proxy` envvar](https://docs.syncthing.net/v1.27.7/users/syncthing.html#proxies).

Code need to be touched located here:

https://github.com/syncthing/syncthing-macos/blob/c1387cb5a6854dc87b9dd49c3971244840eba935/syncthing/STApplication.m#L69

https://github.com/syncthing/syncthing-macos/blob/c1387cb5a6854dc87b9dd49c3971244840eba935/syncthing/STApplication.m#L35

https://github.com/syncthing/syncthing-macos/blob/c1387cb5a6854dc87b9dd49c3971244840eba935/syncthing/DaemonProcess.swift#L63

**Comment by AlliotTech** [comment]
> Yeah I understand you need to bypass the (corporate) firewall or proxy via e.g Tor using the SOCKS5 protocol. For my own reference we need to inject the [`all_proxy` envvar](https://docs.syncthing.net/v1.27.7/users/syncthing.html#proxies).

Yes, I have checked the documentation on environment variables for a specific agent. Is there another way to specify the agent through the application, rather than, as I mentioned earlier, using the command line to start the app to achieve the purpose of injecting environment variables?

**Comment by xor-gate** [comment]
No currently the only environment variable we set/inject is `STNOUPGRADE`. As a intermediate solution we could use the hidden turning knob like settings commandline argument as documented here https://github.com/syncthing/syncthing-macos?tab=readme-ov-file#preferences. We then could add something like:

```
defaults write com.github.xor-gate.syncthing-macosx Proxy 'socks5://127.0.0.1:1234'
```

It would be more nice to have two turning knobs in the GUI so you can set the URL and enable/disable proxying.

I can't promise when I implement this, but its not that hard. Syncthing macOS is a one man show, I do it in my (expensive) free time.

**Comment by xor-gate** [comment]
Maybe we should also detect system wide proxy: https://apple.stackexchange.com/a/396066.

**Comment by AlliotTech** [comment]
> No currently the only environment variable we set/inject is `STNOUPGRADE`. As a intermediate solution we could use the hidden turning knob like settings commandline argument as documented here https://github.com/syncthing/syncthing-macos?tab=readme-ov-file#preferences. We then could add something like:
> 
> ```
> defaults write com.github.xor-gate.syncthing-macosx Proxy 'socks5://127.0.0.1:1234'
> ```
> 
> It would be more nice to have two turning knobs in the GUI so you can set the URL and enable/disable proxying.
> 
> I can't promise when I implement this, but its not that hard. Syncthing macOS is a one man show, I do it in my (expensive) free time.

Thank you sir.
Unfortunately, after tested, the command "defaults write com.github.xor-gate.syncthing-macosx Proxy 'socks5://127.0.0.1:1234'" did not work.

**Comment by xor-gate** [comment]
> Thank you sir. Unfortunately, after tested, the command "defaults write com.github.xor-gate.syncthing-macosx Proxy 'socks5://127.0.0.1:1234'" did not work.

Sure it doesn't work, it was a proposal for myself if that was not clear. It has not been implemented yet to inject the proxy environment variable.
